### DISCUSSION TITLE
Can `flex.amazon.co.uk/download-app` be added to Obtanium? If so, how?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [ImranR98/Obtainium#1047] [Q&A] [answered]
How would I add https://flex.amazon.co.uk/download-app to Obtanium?

**Comment by ImranR98** [comment, marked as answer]
Everything on this page loads through JS, so Obtainium can't scrape it (even if we could do that, the actual download page `https://logistics.amazon.co.uk/app/download-app` looks like it might have some other anti-scraping measures.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/ImranR98/Obtainium/discussions/1047#discussioncomment-7372863

@ImranR98, is anything which might provide a way for Obtainium to access predominantly JS URIs being worked on?

**Reply by ImranR98** [reply]
Not at the moment. I have looked into it before but there don't seem to be any Flutter plugins for Android that can run headless browsers for scraping (which is what would be needed). Something like Selenium or Puppeteer would be needed (Puppeteer exists for Flutter but doesn't support mobile).

**Reply by roke-julian-lockhart** [reply]
> https://github.com/ImranR98/Obtainium/discussions/1047#discussioncomment-7988250

@ImranR98, doesn't even seem like there's an issue for it — https://github.com/puppeteer/puppeteer/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+mobile+flutter returns 0 results. If you see no need to make an issue, I don't mind doing it if it'd be of use, I just have no experience with Flutter though, so I wouldn't be able to answer any follow-ups.

**Reply by ImranR98** [reply]
I don't think that would go anywhere as they've already explained it in the readme:  https://pub.dev/packages/puppeteer#limitations-with-flutter

**Reply by roke-julian-lockhart** [reply]
> https://github.com/ImranR98/Obtainium/discussions/1047#discussioncomment-7988267

Thanks, @ImranR98. Per what you've stated, although I'm aware this might be somewhat stubborn, I've filed an issue at https://github.com/puppeteer/puppeteer/issues/11611#issue-2061682577 which I think is respectful of what they've already said about the issue, *considering there's no other public issue about it*. (If there were, I'd just have subscribed to that.)

**Reply by ImranR98** [reply]
I wonder how apps like NewPipe and Grayjay manage to do it (neither uses Flutter though).

**Reply by roke-julian-lockhart** [reply]
> https://github.com/ImranR98/Obtainium/discussions/1047#discussioncomment-7988336

@ImranR98, https://news.ycombinator.com/item?id=37926438 at least states:

> They embedded a V8 engine to run JavaScript to then make some api calls to the various platforms.

I've not seen any better explanation yet. I expect there's some re-engineering involved, per usual.

**Reply by ImranR98** [reply]
Ah I see. Doing anything like that is definitely beyond me 😂
